### PR TITLE
Added GPS speed to location attributes.

### PR DESCRIPTION
**Pull request by igloochef**
I've added GPS speed to the location attributes to track how fast the device is moving at the specified location, but I'm not sure how useful that is to others.  Please pull if you think it would be useful.

### CODE REVIEW

**Comment by caarmen** [comment]
Hi Igloochef.  Sorry I'm a bit late to read your pull request -- was on vacation.

I tried (quickly) to find some info about the ```speed``` element inside the ```Point``` element in the KML spec, but didn't find anything.  Do you happen to have some documentation about this element?

Thanks!

File: networkmonitor/src/main/java/ca/rmen/android/networkmonitor/provider/NetMonDatabase.java [hunk 1]
```diff
@@ -82,6 +82,7 @@
             + NetMonColumns.DEVICE_LATITUDE + " REAL, "
             + NetMonColumns.DEVICE_LONGITUDE + " REAL, "
             + NetMonColumns.DEVICE_POSITION_ACCURACY + " REAL, "
+            + NetMonColumns.DEVICE_SPEED + " REAL, "
```
Comment (caarmen) [inline networkmonitor/src/main/java/ca/rmen/android/networkmonitor/provider/NetMonDatabase.java]: Missing the upgrade scenario.  See, for example, SQL_UPDATE_TABLE_NETWORKMONITOR_V15_SERVICE_STATE.

Need to update the DATABASE_VERSION to 16.

File: networkmonitor/src/main/res/values/strings_columnnames.xml [hunk 2]
```diff
@@ -57,6 +57,7 @@ limitations under the License.
     <string name="device_latitude">Device Latitude</string>
     <string name="device_longitude">Device Longitude</string>
     <string name="device_position_accuracy">Device Position Accuracy (m)</string>
+    <string name="device_speed">Device Speed (m/s)</string>
```
Comment (caarmen) [inline networkmonitor/src/main/res/values/strings_columnnames.xml]: Missing translations in Spanish and French.
Spanish: ```Velocidad del dispositivo (m/s)```
French: ```Vitesse de l\'appareil (m/s)```

**Comment by igloochef** [comment]
Hello Caarmen,

Thanks for the feedback.  I don't think there is spec in KML for device speed.  My employer needed the information for some tests, so I created the modification for their use.  I'm not really concerned if you deny the pull request as it isn't part of the KML spec.  However, if you would like to include the data, then I can update my branch as per your comments above and then submit another pull request.

Thanks for your feedback!

**Comment by caarmen** [comment]
Hi Igloochef.  I would like to add the device speed field, but without modifying the kml export at all.  The speed should already be included (like all the other fields are), in the ```<ExtendedData>``` element of the Placemarks, or, if you choose the speed as the "field to export", it would appear in the ```<name>``` element of the Placemarks.

I can do the changes, of if you want to do them, and have free time, feel free to submit new commits to the PR.  

In any case, when I end up merging this, I would like to cite you as a contributor, in the about screen.  Please provide your name and a url as you would like them to appear. :)

**Comment by igloochef** [comment]
I've implemented the comments above and put the device speed into &lt;ExtendedData&gt;.
For the contributor screen please use name: Gary Huyser   URL: http://bitaether.net

Let me know if there is anything more, thanks!

File: networkmonitor/src/main/java/ca/rmen/android/networkmonitor/provider/NetMonDatabase.java [hunk 3]
```diff
@@ -191,6 +192,9 @@
     private static final String SQL_UPDATE_TABLE_NETWORKMONITOR_V15_SERVICE_STATE = "ALTER TABLE " + NetMonColumns.TABLE_NAME + " ADD COLUMN "
             + NetMonColumns.SERVICE_STATE + " TEXT";
 
+    private static final String SQL_UPDATE_TABLE_NETWORKMONITOR_V15_DEVICE_SPEED = "ALTER TABLE " + NetMonColumns.TABLE_NAME + " ADD COLUMN "
```
Comment (caarmen) [inline networkmonitor/src/main/java/ca/rmen/android/networkmonitor/provider/NetMonDatabase.java]: This new constant should have V16 instead of V15 in the field name, and it should be executed at the end of ```onUpgrade```, inside a block ```if (oldVersion < 16) {```

File: networkmonitor/src/main/java/ca/rmen/android/networkmonitor/app/dbops/backend/export/kml/KMLWriter.java [hunk 4]
```diff
@@ -73,7 +73,7 @@ public void writeHeader() {
 
     /**
      * Write a single Placemark
-     * 
+     *
```
Comment (caarmen) [inline networkmonitor/src/main/java/ca/rmen/android/networkmonitor/app/dbops/backend/export/kml/KMLWriter.java]: Since KMLWriter only has whitespace changes (after the reverting), could you just revert the changes completely?  One less file in the PR :)

**Comment by caarmen** [comment]
Many thanks for your last commits Igloochef :)

**Comment by caarmen** [comment]
For info, I've done these small changes in a branch on my repo.  I am currently testing it out :)

**Comment by caarmen** [comment]
I'm testing it out, and only getting 0 values for speed.  I haven't actually gone outside to test though... only been testing using simulated movement:
* Fake GPS app
* Loading a GPX file in an emulator
* Genymotion gps commands

Do I have to get my butt off the couch to test this? :)

**Comment by caarmen** [comment]
I found an app which can simulate speed: https://play.google.com/store/apps/details?id=fr.dvilleneuve.lockito&hl=en

I now see non-zero values in netmon for the speed :)

**Comment by igloochef** [comment]
Ha ha yeah, speed actually requires you to move.  We tested it here in a vehicle. :)

**Comment by caarmen** [comment]
A question: are you sure about this url: http://bitaether.net ?
It doesn't seem to be responding.

**Comment by igloochef** [comment]
Yeah, the server it is on is in the process of a network upgrade, it'll be back eventually :)
